lib/elixir/lib/kernel.ex

```diff
@@ -2973,7 +2973,9 @@ defmodule Kernel do
 
   translates to:
 
-      when x >= 1 and x <= 3
+      when is_integer(x) and x >= 1 and x <= 3
+
+  Note that a value can only be `in` a range when it's an integer
```

Comment:
Potentially this whole line could be removed since this is implied by the now added `is_integer(x)`